kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/ConflictChecker.java

```diff
@@ -113,7 +125,10 @@ public TransactionRebaseState resolveConflicts(Engine engine) throws ConcurrentW
 
         // if we get here, we have successfully rebased (i.e no logical conflicts)
         // against the winning transactions
-        return new TransactionRebaseState(getLastWinningTxnVersion(winningCommits));
+        return new TransactionRebaseState(
+                lastWinningVersion,
+                getLastCommitTimestamp(
+                        engine, lastWinningVersion, winningCommits, winningCommitInfoOpt.get()));
```

Comment:
Do you want to check that `winningCommitInfoOpt` is not empty?

Reply:
This get is for the atomic reference but not for the optional.

Reply:
It starts with `optional.empty`. And in the [getLastCommitTimestamp](https://github.com/delta-io/delta/blob/de0155896cdacd64d8ec87aa260f2e38a002b62e/kernel/kernel-api/src/main/java/io/delta/kernel/internal/replay/ConflictChecker.java#L275) function it will check if it's ict enabled. And if it's ict enabled and the winningCommitInfoOpt is empty, the `CommitInfo.getRequiredInCommitTimestamp` will raise an error.